pkg/cloudprovider/providers/azure/azure_backoff.go

```diff
@@ -262,7 +262,14 @@ func (az *Cloud) CreateOrUpdateRouteTableWithRetry(routeTable network.RouteTable
 		ctx, cancel := getContextWithCancel()
 		defer cancel()
 
-		resp, err := az.RouteTablesClient.CreateOrUpdate(ctx, az.ResourceGroup, az.RouteTableName, routeTable)
+		var rg string
+		if len(az.RouteTableResourceGroup) > 0 {
```

Comment:
The defaulting should be done outside, as this will avoid duplication

Reply:
Changed !